contracts/bonding/BondingManager.sol

```diff
@@ -165,41 +165,76 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         whenSystemNotPaused
         currentRoundInitialized
     {
-        // Only callable if current round is not locked
-        require(!roundsManager().currentRoundLocked());
-        // Block reward cut must be a valid percentage
-        require(MathUtils.validPerc(_blockRewardCut));
-        // Fee share must be a valid percentage
-        require(MathUtils.validPerc(_feeShare));
-
+        Transcoder storage t = transcoders[msg.sender];
         Delegator storage del = delegators[msg.sender];
 
-        // Must have a non-zero amount bonded to self
-        require(del.delegateAddress == msg.sender && del.bondedAmount > 0);
+        if (roundsManager().currentRoundLocked()) {
+            // If it is the lock period of the current round
+            // the lowest price previously set by any transcoder
+            // becomes the price floor and the caller can lower its
+            // own price to a point greater than or equal to the price floor
+
+            // Caller must already be a registered transcoder
+            require(transcoderStatus(msg.sender) == TranscoderStatus.Registered);
```

Comment:
should we also require that `_feeShare` and `_blockRewardCut` are not different than the pending values? I'm not sure...either the values won't update silently which could be weird, or else they risk wasting gas if they don't realize they have to set these exactly the same.

Reply:
Hm yeah I do think that ignoring the values silently is a bit weird...I would lean towards adding the require checks for `feeShare` and `blockRewardCut` to make the intent of the function call clear: "I am leaving my feeShare and blockRewardCut unchanged by providing the same values as the pending values and changing the pricePerSegment" instead of "I am  providing arbitrary values for feeShare and blockRewardCut that will be ignored". I think the defenses against a user accidentally providing incompatible feeShare and blockRewardCut values can be added in the client